### ISSUE TITLE
Configuring kafka.message.iterator.className in kafka-2.0.0 profile

### ISSUE THREAD

**Issue by ankon** [pinterest/secor#461]
The `com.pinterest.secor.reader.LegacyKafkaMessageIterator` is not included when using `mvn -Pkafka-2.0.0`, but it is still mentioned in the default `secor.common.properties`. Due to the way Apache Commons PropertiesConfiguration works it is not possible (?) to _overwrite_ this setting, but rather setting in a configuration that includes `secor.common.properties` directly or indirectly it will just append the value, leading to:
~~~~
Nov 09, 2018 3:06:38 PM com.twitter.ostrich.admin.BackgroundProcess start
INFO: Starting LatchedStatsListener
Nov 09, 2018 3:06:38 PM com.twitter.ostrich.admin.AdminHttpService start
INFO: Admin HTTP interface started on port 9998.
2018-11-09 15:06:38,255 [main] (com.pinterest.secor.common.OstrichAdminService:68) INFO  build.properties build_revision: 7d253c1a5ad1d438990e0362785a7a42346ce625
2018-11-09 15:06:38,267 [main] (com.pinterest.secor.main.ConsumerMain:69) INFO  starting 7 consumer threads
2018-11-09 15:06:38,271 [Thread-6] (com.pinterest.secor.main.ConsumerMain$1:65) ERROR Thread Thread[Thread-6,5,main] failed
java.lang.RuntimeException: Failed to initialize the consumer
	at com.pinterest.secor.consumer.Consumer.run(Consumer.java:90)
Caused by: java.lang.RuntimeException: java.lang.ClassNotFoundException: com.pinterest.secor.reader.LegacyKafkaMessageIterator
	at com.pinterest.secor.reader.KafkaMessageIteratorFactory.getIterator(KafkaMessageIteratorFactory.java:15)
	at com.pinterest.secor.reader.MessageReader.<init>(MessageReader.java:58)
	at com.pinterest.secor.consumer.Consumer.init(Consumer.java:69)
	at com.pinterest.secor.consumer.Consumer.run(Consumer.java:88)
Caused by: java.lang.ClassNotFoundException: com.pinterest.secor.reader.LegacyKafkaMessageIterator
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:264)
	at com.pinterest.secor.reader.KafkaMessageIteratorFactory.getIterator(KafkaMessageIteratorFactory.java:11)
	... 3 more
2018-11-09 15:06:38,271 [Thread-9] (com.pinterest.secor.main.ConsumerMain$1:65) ERROR Thread Thread[Thread-9,5,main] failed
java.lang.RuntimeException: Failed to initialize the consumer
	at com.pinterest.secor.consumer.Consumer.run(Consumer.java:90)
Caused by: java.lang.RuntimeException: java.lang.ClassNotFoundException: com.pinterest.secor.reader.LegacyKafkaMessageIterator
	at com.pinterest.secor.reader.KafkaMessageIteratorFactory.getIterator(KafkaMessageIteratorFactory.java:15)
	at com.pinterest.secor.reader.MessageReader.<init>(MessageReader.java:58)
	at com.pinterest.secor.consumer.Consumer.init(Consumer.java:69)
	at com.pinterest.secor.consumer.Consumer.run(Consumer.java:88)
Caused by: java.lang.ClassNotFoundException: com.pinterest.secor.reader.LegacyKafkaMessageIterator
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:264)
	at com.pinterest.secor.reader.KafkaMessageIteratorFactory.getIterator(KafkaMessageIteratorFactory.java:11)
	... 3 more
~~~~

From https://commons.apache.org/proper/commons-configuration/apidocs/org/apache/commons/configuration2/PropertiesConfiguration.html:
> If a key is used more than once, the values are appended like if they were on the same line separated with commas. Note: When the configuration file is written back to disk the associated PropertiesConfigurationLayout object (see below) will try to preserve as much of the original format as possible, i.e. properties with multiple values defined on a single line will also be written back on a single line, and multiple occurrences of a single key will be written on multiple lines. If the addProperty() method was called multiple times for adding multiple values to a property, these properties will per default be written on multiple lines in the output file, too. Some options of the PropertiesConfigurationLayout class have influence on that behavior.

Right now it seems that either I need to duplicate the default configuration completely, or apply magic[1].

--
[1] In my case "magic" will be to apply a local patch to the `secor.common.properties` file to set `kafka.message.iterator.className` and `kafka.client.className` before building.

**Comment by ankon** [comment]
So, it seems that that is actually a generic issue in the configuration, for example it's not possible to easily set `secor.file.reader.writer.factory=com.pinterest.secor.io.impl.DelimitedTextFileReaderWriterFactory` either.

What I found working now is to first set my values, and only thereafter doe the include:
```
secor.file.reader.writer.factory=com.pinterest.secor.io.impl.DelimitedTextFileReaderWriterFactory
include=secor.prod.properties
```

Presumably the same would work with the mentioned `kafka.message.iterator.className` and `kafka.client.className` properties.

I see that the README indicates to _modify_ the provided configuration files, so that's likely "fine". It does lead to issues when trying to treat the secor distribution as "given", and provide just overrides/customizations in a separate configuration file.

**Comment by vickykumarmlt** [comment]
@ankon am getting the same exception while running secor. can you help me with that?

2022-08-31 20:41:49,380 [Thread-11] (com.pinterest.secor.consumer.Consumer) ERROR Thread failed
java.lang.RuntimeException: Failed to initialize the consumer
	at com.pinterest.secor.consumer.Consumer.run(Consumer.java:155)
Caused by: java.lang.RuntimeException: java.lang.ClassNotFoundException: com.pinterest.secor.reader.LegacyKafkaMessageIterator
	at com.pinterest.secor.reader.KafkaMessageIteratorFactory.getIterator(KafkaMessageIteratorFactory.java:33)
	at com.pinterest.secor.consumer.Consumer.init(Consumer.java:93)
	at com.pinterest.secor.consumer.Consumer.run(Consumer.java:152)
Caused by: java.lang.ClassNotFoundException: com.pinterest.secor.reader.LegacyKafkaMessageIterator
	at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:352)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:264)
	at com.pinterest.secor.reader.KafkaMessageIteratorFactory.getIterator(KafkaMessageIteratorFactory.java:29)

**Comment by ankon** [comment]
@vickykumarmlt I am no longer using Secor (or kafka, or any kafka-based environment, for that matter), so unfortunately no, I cannot help. This issue was closed as fixed (see #463), and in the 4 years since then a lot has likely changed. I suggest you open a new issue to increase the chance of more people seeing your issue and maybe offering their help.
